Src/FluentAssertions/Common/StringExtensions.cs

```diff
@@ -7,6 +8,23 @@ namespace FluentAssertions.Common;
 
 internal static class StringExtensions
 {
+    /// <summary>
+    /// Finds the first index at which the <paramref name="value"/> does not match the <paramref name="expected"/>
+    /// string anymore, accounting for the specified <paramref name="comparer"/>.
+    /// </summary>
+    public static int IndexOfFirstMismatch(this string value, string expected, IEqualityComparer<string> comparer)
+    {
+        for (int index = 0; index < value.Length; index++)
+        {
+            if (index >= expected.Length || !comparer.Equals(value[index..(index + 1)], expected[index..(index + 1)]))
+            {
+                return index;
+            }
+        }
+
+        return -1;
```

Comment:
🔧 This line is not covered by a test.

Reply:
Really? I changed the value to `-2` and then got 352 failing tests:
![image](https://github.com/fluentassertions/fluentassertions/assets/3438234/d75da02a-bbf4-4153-8790-9655c429423a)

Reply:
Ah, I see. It is line 44 that is no longer covered...
Will check it right away :-)

Reply:
I found a way to remove the previous overload of `IndexOfFirstMismatch` to simplify the implementation.